contrib/completion/bash/docker

```diff
@@ -201,19 +201,36 @@ __docker_signals() {
 
 _docker_docker() {
 	local boolean_options="
-		--daemon -d
 		--debug -D
 		--help -h
+		--tls
+		--tlsverify
+		--tlscacert
+		--tlscert
+		--tlskey
+		--version -v
+	"
+
+	case "$cur" in
+		-*)
+			COMPREPLY=( $( compgen -W "$boolean_options $main_options_with_args" -- "$cur" ) )
+			;;
+		*)
+			COMPREPLY=( $( compgen -W "${commands[*]} help" -- "$cur" ) )
+			;;
+	esac
+}
+
+_docker_daemon() {
+	local boolean_options="
+		--help -h
 		--icc
 		--ip-forward
 		--ip-masq
 		--iptables
 		--ipv6
 		--selinux-enabled
-		--tls
-		--tlsverify
 		--userland-proxy=false
```

Comment:
I suggest to also modify completion for the others boolean options defaulting to `true` so that they also complete to `--xxx=false`:
```
--icc --ip-forward --ip-masq --iptables
```